### PR TITLE
Longw/lcm conformance

### PR DESCRIPTION
**Pull request by wanlonghenry**

### CODE REVIEW

File: test/e2e/src/core/e2e_tests.sh [hunk 1]
```diff
@@ -51,18 +51,67 @@ waitForArcK8sClusterCreated() {
...
+    connectivityState=false
+    max_retries=60
+    sleep_seconds=10
+    for i in $(seq 1 $max_retries)
+    do
+      echo "iteration: ${i}, clustername: ${CLUSTER_NAME}, resourcegroup: ${RESOURCE_GROUP}"
+      ## hybridaks does not support checking connectivityStatus  maybe compare date with system date to see if has difference
+      # az hybridaks show -n $k8sClusterName -g $resourceGroup --query properties.status.featuresStatus.arcAgentStatus.lastConnectivityTime -o json
+
+      clusterState=$(az connectedk8s show --name $CLUSTER_NAME --resource-group $RESOURCE_GROUP --query connectivityStatus -o json)  arc
```
Comment (ganga1980) [inline test/e2e/src/core/e2e_tests.sh:64]: i see "arc"  in line 64.  looks like its typo.

Comment (wanlonghenry) [reply test/e2e/src/core/e2e_tests.sh:64]: this one currently the hybridaks command does not support the connectivityStatus parameter, and this query needs to  be replaced by the hybridaks command to check the cluster status

Comment (ganga1980) [reply test/e2e/src/core/e2e_tests.sh:64]: I am referring "arc"  at the end of the line.

File: test/e2e/src/tests/test_ds_workflows.py [hunk 2]
```diff
@@ -27,6 +27,7 @@ def test_ds_workflows(env_dict):
     api_instance = client.CoreV1Api()
 
     daemonsetPodLabelSelector = constants.AGENT_DAEMON_SET_PODS_LABEL_SELECTOR
+    ### dict get
```
Comment (ganga1980) [inline test/e2e/src/tests/test_ds_workflows.py:30]: I dont see any change other this comment and if so, please revert both the test files.

File: test/e2e/src/core/e2e_tests.sh [hunk 3]
```diff
@@ -51,18 +51,67 @@ waitForArcK8sClusterCreated() {
...
 waitForCIExtensionInstalled() {
     installedState=false
     max_retries=60
     sleep_seconds=10
     for i in $(seq 1 $max_retries)
     do
       echo "iteration: ${i}, clustername: ${CLUSTER_NAME}, resourcegroup: ${RESOURCE_GROUP}"
-      installState=$(az k8s-extension show  --cluster-name $CLUSTER_NAME --resource-group $RESOURCE_GROUP  --cluster-type connectedClusters --name azuremonitor-containers --query installState -o json)
+      installState=$(az k8s-extension show --cluster-name $CLUSTER_NAME --resource-group $RESOURCE_GROUP  --cluster-type connectedClusters --name azuremonitor-containers --query provisioningState -o json)
       installState=$(echo $installState | tr -d '"' | tr -d '"\r\n')
```
Comment (ganga1980) [inline test/e2e/src/core/e2e_tests.sh:86]: why this has been changed from installState to provisioningState for connectedClusters?

File: test/e2e/src/core/e2e_tests.sh [hunk 4]
```diff
@@ -51,18 +51,67 @@ waitForArcK8sClusterCreated() {
...
+
+waitForLCMCIExtensionInstalled() {
+    installedState=false
+    max_retries=60
+    sleep_seconds=10
+    for i in $(seq 1 $max_retries)
+    do
+      echo "iteration: ${i}, clustername: ${CLUSTER_NAME}, resourcegroup: ${RESOURCE_GROUP}"
+      ### for lcm, replace command with provisioned clusted
+      # installed version  installState missing
```
Comment (ganga1980) [inline test/e2e/src/core/e2e_tests.sh:107]: until we get the installState support, you can use provisioningState  and add few seconds of sleep to get extension installed.

File: test/e2e/src/core/e2e_tests.sh (hunk 3 above)
Comment (wanlonghenry) [reply test/e2e/src/core/e2e_tests.sh:86]: looks like the k8s-extension team has removed the installState in the parameters, and I also checked that in the portal/ui side, it is using the provisioningState to check the extension status

**Pull request closed without merging**